S3Proxy/cache.go

```diff
@@ -0,0 +1,49 @@
+package S3Proxy
+
+import (
+	"encoding/json"
+	"time"
+)
+
+var s3Buckets = map[string]*s3BucketCacheItem{}
+
+type s3BucketCacheItem struct {
+	Name      string
+	Location  string
+	Timestamp time.Time
+	TTL       time.Duration
+}
+
+func (b s3BucketCacheItem) String() string {
+	tmp := &b
+	tmp.TTL = b.TTL - time.Since(b.Timestamp)
+	out, _ := json.Marshal(tmp)
+	return string(out)
```

Comment:
Cool, you can go ahead and merge those changes in.

On Wed, Apr 22, 2015 at 10:53 PM, Quiran Storey <notifications@github.com>
wrote:

> In S3Proxy/cache.go
> <https://github.com/qstorey/S3Proxy/pull/8#discussion_r28913657>:
>
> > +)
> > +
> > +var s3Buckets = map[string]*s3BucketCacheItem{}
> > +
> > +type s3BucketCacheItem struct {
> > +	Name      string
> > +	Location  string
> > +	Timestamp time.Time
> > +	TTL       time.Duration
> > +}
> > +
> > +func (b s3BucketCacheItem) String() string {
> > +	tmp := &b
> > +	tmp.TTL = b.TTL - time.Since(b.Timestamp)
> > +	out, _ := json.Marshal(tmp)
> > +	return string(out)
>
> Just confirm that what I did above here is what you were talking about.
> The unit tests still pass, but I'm not sure if this is the implementation
> you were talking about.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/qstorey/S3Proxy/pull/8/files#r28913657>.
>